partcad/src/partcad/ai_feature_file.py

```
@@ -0,0 +1,92 @@
+        self.url = url
+        self.is_text = False
+        self.is_image = False
+        self.is_video = False
+        self.is_pdf = False
+        self.suffix = ".txt"
+        self.filename = None
+
+        # Switch to lowercase to normalize the file extension in the URL
+        lower_url = url.lower()
+        if lower_url.startswith("http://") or lower_url.startswith("https://"):
+            response = requests.get(self.url)
```

Ruff (0.8.2)

Probable use of `requests` call without timeout

(S113)